server/src/structs.rs

```diff
@@ -0,0 +1,231 @@
+pub struct Location {
+    pub latitude: OrderedFloat<f64>,
+    pub longitude: OrderedFloat<f64>,
+}
+
+// =====================================================================
+// Asset types
+// =====================================================================
+
+/// A struct representing an aircraft.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Aircraft {
+    pub id: Uuid,
+    pub name: Option<String>,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
+    pub created_at: SystemTime,
+    pub updated_at: Option<SystemTime>,
+    /// If the vector is empty, the aircraft is available to everyone.
+    /// Otherwise, the aircraft is only available to the clients in the
+    /// vector.
+    pub restricted_to: Vec<Uuid>,
+    pub status: AssetStatus,
+    /// The aircraft's manufacturer.
+    ///
+    /// TODO: For now we can just say "Boeing", "Airbus", etc. Later, we
+    /// can a struct for this and store the manufacturer's name, logo,
+    /// etc.
+    pub manufacturer: String,
+    pub model: String,
+    pub serial_number: String,
+    /// The aircraft's registration number.
+    ///
+    /// In the US, this is the N number.
+    ///
+    /// This is a unique identifier for
+    /// the aircraft that can be used to look up information about the
+    /// aircraft from national aviation authorities like the FAA.
+    pub registration_number: String,
+    pub description: Option<String>,
+    pub max_payload: OrderedFloat<f32>,
+    pub max_range: Option<OrderedFloat<f64>>,
+}
+
+impl Aircraft {
+    /// Get the aircraft's name.
+    ///
+    /// If the aircraft does not have a name, this will be `manufacturer
+    /// model registration_number`.
+    ///
+    /// For example, "Boeing 737-800 N12345".
+    pub fn full_name(&self) -> String {
+        match &self.name {
+            Some(name) => name.clone(),
+            None => format!(
+                "{} {} {}",
+                self.manufacturer, self.model, self.registration_number
+            ),
+        }
+    }
+}
+
+/// A struct representing a vertipad (a vertical landing pad).
+///
+/// A vertipad is a landing pad that is used for vertical takeoff and
+/// landing (VTOL) aircraft. Usually, vertipads belong to vertiports.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Vertipad {
+    pub id: Uuid,
+    pub vertiport_id: Uuid,
+    pub status: AssetStatus,
+    pub location: Location,
+}
+
+/// A struct representing a vertiport (a vertical airport).
+///
+/// A vertiport is an airport that is used for vertical takeoff and
+/// landing (VTOL) aircraft. A vertiport may have one or more vertipads.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Vertiport {
+    pub id: Uuid,
+    pub name: String,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
```

Comment:
Same comment about groups being delegated, not individual assets

Reply:
[21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)